src/ImageSharp/Formats/Jpg/Components/Block.cs

```diff
@@ -37,8 +68,149 @@ public Block()
         /// </returns>
         public int this[int index]
         {
-            get { return this.data[index]; }
-            set { this.data[index] = value; }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get { return this.Data[index]; }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            set { this.Data[index] = value; }
+        }
+
+        // TODO: Refactor Block.Dispose() callers to always use 'using' or 'finally' statement!
+        public void Dispose()
+        {
+            if (Data != null)
+            {
+                ArrayPool.Return(Data, true);
+                Data = null;
+            }
+        }
+
+        public static void DisposeAll(Block[] blocks)
+        {
+            for (int i = 0; i < blocks.Length; i++)
+            {
+                blocks[i].Dispose();
+            }
+        }
+
+
+        public void Clear()
+        {
+            for (int i = 0; i < Data.Length; i++)
+            {
+                Data[i] = 0;
+            }
+        }
+
+        public Block Clone()
+        {
+            Block clone = Create();
+            Array.Copy(Data, clone.Data, BlockSize);
+            return clone;
+        }
+    }
+
+    /// <summary>
+    /// Temporal class to make refactoring easier.
+    /// 1. Refactor Block -> BlockF
+    /// 2. Test
+    /// 3. Refactor BlockF -> Block8x8F
+    /// </summary>
+    internal struct BlockF : IDisposable
```

Comment:
Strictly speaking: not. But: JpegEncoderCore still has to be refactored to use Block8x8F, and it's more safe to do it in 2 steps, as described in the comment, I think it would be better to remove it, when JpegEncoderCore is refactored to use Block8x8F.